addons/repair/functions/fnc_addRepairActions.sqf

```diff
@@ -112,7 +126,11 @@ _wheelHitPointSelections = _wheelHitPointsWithSelections select 1;
         } else {
             private "_action";
             _action = [_name, _text, _icon, _statement, _condition, {}, [_x, "MiscRepair"], _selection, 4] call EFUNC(interact_menu,createAction);
-            [_type, 0, ["ACE_MainActions", QGVAR(Repair)], _action] call EFUNC(interact_menu,addActionToClass);
+            [   _type,
+                0,
+                if (_selection isEqualTo [0, 0 ,0]) then { ["ACE_MainActions", QGVAR(Repair)] } else { [] }, // Put inside main actions if no other position was found above
```

Comment:
Agreed, would be most clean if that check would be made in `addActionToClass` and `addActionToObject` functions though.